magma/syntax/sequential.py

```diff
@@ -26,6 +26,8 @@ def visit_Attribute(self, node):
                 return ast.Name(f"self_{node.attr}_I", ast.Store())
             else:
                 return ast.Name(f"self_{node.attr}_O", ast.Load())
+        else:
+            self.generic_visit(node)
         return node
```

Comment:
I think this relies on the `generic_visit` modifying the node in place which isn't always guaranteed.  Can you update this to remove the `else` and say
```
return self.generic_visit(node)
```

that we always recurse if we're not doing the rewrite behavior and return the result.

Reply:
sure, updated